modules/json/src/smithy4s/http/json/SchemaVisitorJCodec.scala

```diff
@@ -1205,7 +1220,11 @@ private[smithy4s] class SchemaVisitorJCodec(maxArity: Int)
               if (f.isRequired) {
                 if (value == null) cursor.requiredFieldError(f.label, f.label)
                 value
-              } else Option(value)
+              } else {
+                Option(value).orElse {
+                  Option(defaults.get(f.label))
+                }
+              }
```

Comment:
Not sure if orElse is as efficient here as other methods

Reply:
it's not, because of the cost of the Option allocation. Also you're only leveraging defaults values on optional fields there. 

It should be : 

``` 
fieldsWithDefaults.foreach { case (f, default) => 
 ....
  if (f.isRequired) {
      if(default == null) {
      ... 
      }  else {
        default 
      }
  } 
} 
```

Reply:
The reason I put it in the optional side is that smithy doesn't allow required fields to have default values (required and default traits are mutually exclusive). But I do get your broader point and will make the update

Reply:
@lewisjkl actually that's an interesting notion : in the smithy4s level, I think the presence of `@default` should translate to a `required` field, and in the case class, the field should not be wrapped in Option. Essentially `@default` means that an instance of the case class always has a value for the corresponding field, which means in the Schema, the field should be considered required.

Reply:
Ah, I see what you're saying. I will make that change 👍